community/kernel/src/main/java/org/neo4j/kernel/impl/core/GraphPropertiesProxy.java

```diff
@@ -195,6 +198,45 @@ public Object removeProperty( String key )
     }
 
     @Override
+    public Map<String, Object> getProperties( String... names )
+    {
+        try ( Statement statement = actions.statement() )
+        {
+            Map<String, Object> properties = new HashMap<>();
+            ReadOperations readOperations = statement.readOperations();
+            for ( String name : names )
+            {
+                int propertyKeyId = readOperations.propertyKeyGetForName( name );
+                properties.put( name, readOperations.graphGetProperty( propertyKeyId ) );
```

Comment:
I would prefer if only found properties were included, in all cases. That would match better the semantics that we already have in place, i.e. there cannot be null valued properties.